protostar/commands/test/test_command.py

```diff
@@ -213,3 +226,30 @@ async def test(
                 )
 
             return testing_summary
+
+    def _log_test_collector_result(
+        self, test_collector_result: TestCollector.Result
+    ) -> None:
+        for broken_test_suite in test_collector_result.broken_test_suites:
+            self._log_formatted_test_result(broken_test_suite)
+        if test_collector_result.test_cases_count:
+            self._log_formatted_test_collector_summary(test_collector_result)
+        else:
+            self._logger.warning("No test cases found")
+
+    def _log_formatted_test_collector_summary(
+        self, test_collector_result: TestCollector.Result
+    ) -> None:
+        formatter_view_model = (
+            TestCollectorSummaryFormatter.ViewModel.from_test_result_summary(
+                test_collector_result
+            )
+        )
+        formatted_result = self._test_collector_summary_formatter.format(
+            formatter_view_model
+        )
+        self._logger.info(formatted_result)
+
+    def _log_formatted_test_result(self, test_result: TestResult) -> None:
+        formatted_test_result = self._test_result_formatter.format(test_result)
+        print(formatted_test_result)
```

Comment:
Nope. The logger info attaches TIME [INFO] prefix, which is not desirable in the testing output. Logger is quite hard to configure (Protostar should've used `LoggerFacade` instead from the beginning), so I left the `print` as it was.

Reply:
Then why is collector summary printed to the logger?

> (Protostar should've used `LoggerFacade` instead from the beginning)

I do not agree, I think we are simply mixing logging with user interface here.

Reply:
> Then why is collector summary printed to the logger?

To be consistent. Logger is used across the app to output this kind of output to the user. Except for exceptions and test results because logging style doesn't work well with them. That's why I wanted to add a facade to have better control over formatting.

If I use logger, the output looks like this:

![Screen Shot 2022-08-16 at 19 54 13](https://user-images.githubusercontent.com/9629577/184948204-320841fb-d1fd-44cd-ab7f-87d38fa34c83.png)

Reply:
> To be consistent. Logger is used across the app to output this kind of output to the user. Except for exceptions and test results because logging style doesn't work well with them. That's why I wanted to add a facade to have better control over formatting.

Let's leave the code as-is in this PR, but I would like to start us thinking if we aren't doing the things badly here. I think that all user messages should be `print`-ed and only Protostar "internal" messages should be logged.

Reply:
>  I would like to start us thinking if we aren't doing the things badly here

Yep, something is wrong here.

> I think that all user messages should be print-ed and only Protostar "internal" messages should be logged

Logger unlike `print` can control verboseness. I'd probably create `ProtostarLogger` tailored for the application needs. `ProtostarLogger` would use print under the hood.

Reply:
> Logger unlike `print` can control verboseness.

But we don't have any verboseness switches, after all.